### PR TITLE
Equativ Bid Adapter: initial release, Smartadserver Bid Adapter: take lowest floor to send to endpoint

### PR DESCRIPTION
**Pull request by krzysztofequativ**
## Type of change
- [x] New bidder adapter
- [x] Feature

### CODE REVIEW

**Comment by krzysztofequativ** [comment]
@patmmccann, could you share when the review could be perform, please? We have a pilot publisher for the new adapter and would like to schedule the work. Thanks in advance for your feedback.

**Comment by patmmccann** [comment]
Review feedback on slack

File: modules/equativBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  },
+
+  imp(buildImp, bidRequest, context) {
+    const imp = buildImp(bidRequest, context);
+    const { siteId, pageId, formatId } = bidRequest.params;
+
+    delete imp.dt;
+
+    imp.bidfloor = imp.bidfloor || spec.getMinFloor(bidRequest);
+    imp.secure = Number(window.location.protocol === 'https:');
```
Comment (patmmccann) [inline modules/equativBidAdapter.js:109]: Window doesn't matter, the safeframe is always secure

File: modules/equativBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+      method: 'POST',
+      url: 'https://ssb-global.smartadserver.com/api/bid?callerId=169'
+    };
+  },
+
+  /**
+   * @param bidRequest
+   * @returns {number}
+   */
+  getMinFloor: (bidRequest) => {
```
Comment (patmmccann) [inline modules/equativBidAdapter.js:33]: You guys have this same code block in two adapters, please import it

File: modules/smartadserverBidAdapter.js [hunk 3]
```diff
@@ -332,18 +331,22 @@ export const spec = {
...
     }
 
-    const floor = bid.getFloor({
-      currency: currency || 'USD',
-      mediaType,
-      size: '*'
-    });
-
-    return isPlainObject(floor) && !isNaN(floor.floor) ? floor.floor : DEFAULT_FLOOR;
+    return floors.length ? Math.min(...floors) : DEFAULT_FLOOR;
```
Comment (patmmccann) [inline modules/smartadserverBidAdapter.js:349]: line 357 below is throwing a warning:

/home/circleci/Prebid.js/modules/smartadserverBidAdapter.js
  357:1  warning  The type 'syncs' is undefined  jsdoc/no-undefined-types

File: modules/equativBidAdapter.js (hunk 1 above)
Comment (krzysztofequativ) [reply modules/equativBidAdapter.js]: Updated. However, I see many other adapters which calculate the secure property in the same way. Does it mean that they also should be updated?

File: modules/equativBidAdapter.js (hunk 2 above)
Comment (krzysztofequativ) [reply modules/equativBidAdapter.js]: Done.

File: modules/smartadserverBidAdapter.js (hunk 3 above)
Comment (krzysztofequativ) [reply modules/smartadserverBidAdapter.js]: Fixed.

**Comment by krzysztofequativ** [comment]
Thanks @patmmccann for your immediate response. I've fixed all the mentioned points. Please continue reviewing.

File: modules/equativBidAdapter.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+      deepSetValue(req, 'site.publisher.id', bid.ortb2.site.publisher.id || bid.params.networkId);
+    } else if (deepAccess(bid, 'ortb2.app.publisher')) {
+      deepSetValue(req, 'app.publisher.id', bid.ortb2.app.publisher.id || bid.params.networkId);
+    } else if (deepAccess(bid, 'ortb2.dooh.publisher')) {
+      deepSetValue(req, 'dooh.publisher.id', bid.ortb2.dooh.publisher.id || bid.params.networkId);
+    } else {
+      deepSetValue(req, 'site.publisher.id', bid.params.networkId);
+    }
+
+    const pid = storage.getCookie(PID_COOKIE_NAME);
```
Comment (patmmccann) [inline modules/equativBidAdapter.js:137]: why not check local storage?

Comment (krzysztofequativ) [reply modules/equativBidAdapter.js:137]: Indeed, we're going to improve in the next update.

**Pull request merged**